test-jest/chart/RadialBarChart.spec.tsx

```diff
@@ -0,0 +1,176 @@
+        cx={150}
+        cy={150}
+        innerRadius={20}
+        outerRadius={140}
+        barSize={10}
+        data={data}
+      >
+        <RadialBar startAngle={0} endAngle={180} label={label} background dataKey="uv" isAnimationActive={false} />
+      </RadialBarChart>,
+    );
+    expect(container.querySelectorAll('.recharts-radial-bar-sector')).toHaveLength(7);
+  });
+
+  test('Renders 7 customized shape when shape is set to be a react element', () => {
+    const label = { orientation: 'outer' };
+    const Shape = (props: any) => <Sector {...props} className="customized-shape" />;
+    const { container } = render(
+      <RadialBarChart
+        width={500}
+        height={300}
+        cx={150}
+        cy={150}
+        innerRadius={20}
+        outerRadius={140}
+        barSize={10}
+        data={data}
+      >
+        <RadialBar shape={<Shape />} label={label} background dataKey="uv" isAnimationActive={false} />
+      </RadialBarChart>,
+    );
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(7);
+  });
+
+  test('Renders 7 customized shape when shape is set to be a function', () => {
+    const label = { orientation: 'outer' };
+    const renderShape = (props: any) => <Sector {...props} className="customized-shape" />;
+    const { container } = render(
+      <RadialBarChart
+        width={500}
+        height={300}
+        cx={150}
+        cy={150}
+        innerRadius={20}
+        outerRadius={140}
+        barSize={10}
+        data={data}
+      >
+        <RadialBar shape={renderShape} label={label} background dataKey="uv" isAnimationActive={false} />
+      </RadialBarChart>,
+    );
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(7);
+  });
+
+  test('Renders 7 legend item when add a Legend element', () => {
+    const { container } = render(
+      <RadialBarChart
+        width={500}
+        height={300}
+        cx={150}
+        cy={150}
+        innerRadius={20}
+        outerRadius={140}
+        barSize={10}
+        data={data}
+      >
+        <RadialBar background dataKey="uv" isAnimationActive={false} />
+        <Legend />
+      </RadialBarChart>,
+    );
+    expect(container.querySelectorAll('.recharts-legend-wrapper')).toHaveLength(1);
+    expect(container.querySelectorAll('.recharts-legend-item')).toHaveLength(7);
+  });
+
+  test('Renders tooltip when add a Tooltip element', () => {
+    const { container } = render(
+      <RadialBarChart
+        width={500}
+        height={300}
+        cx={150}
+        cy={150}
+        innerRadius={20}
+        outerRadius={140}
+        barSize={10}
+        data={data}
+      >
+        <RadialBar background dataKey="uv" isAnimationActive={false} />
+        <Tooltip />
+      </RadialBarChart>,
+    );
+
+    expect(container.querySelectorAll('.recharts-tooltip-wrapper')).toHaveLength(1);
+    expect(container.querySelectorAll('.recharts-default-tooltip')).toHaveLength(1);
```

Comment:
Interesting, does this work without firing events? Good enough for now though, thank you!

Reply:
Thank you!

Reply:
The wrapper is technically always there it just isn't shown/doesn't have content